src/relay/pass/type_infer.cc

```diff
@@ -532,10 +571,19 @@ Expr TypeInferencer::Infer(Expr expr) {
 }
 
 
-Expr InferType(const Expr& expr, const Module& mod) {
-  auto e = TypeInferencer(mod).Infer(expr);
-  CHECK(WellFormed(e));
-  return e;
+Expr InferType(const Expr& expr, const Module& mod_ref) {
+  Module mod = mod_ref;
```

Comment:
Won't this make an unnecessary copy of `mod_ref`?

Reply:
It will copy the outer container, the code was written slightly differently before, will change.